pkgs/development/compilers/jetbrains-jdk/default.nix

```diff
@@ -1,16 +1,20 @@
-{ lib, openjdk11, fetchFromGitHub, jetbrains }:
+{ lib
+, fetchFromGitHub
+, jetbrains
+, openjdk17
+}:
 
-openjdk11.overrideAttrs (oldAttrs: rec {
+openjdk17.overrideAttrs (oldAttrs: rec {
   pname = "jetbrains-jdk";
-  version = "11_0_13-b1751.25";
+  version = "17.0.3-b469.32";
 
   src = fetchFromGitHub {
     owner = "JetBrains";
     repo = "JetBrainsRuntime";
     rev = "jb${version}";
-    sha256 = "sha256-TPNYZUkAoiZfp7Ci3fslKnRNGY1lnyIhXYUt6J31lwI=";
+    hash = "sha256-WKgNo4eFOCl0tN6LhBsgxzbD355Wc50WBk1mXwmEuag=";
   };
```

Comment:
> There's a hash mismatch on Darwin too, see ofborg.

Hi, confused about this @Mindavi, it is from a git fetch with a specific revision. :O

A hash mismatch shouldn't be possible to begin with (specially since the linux x86_64 / aarch64 jobs succeeded.

Reply:
It might be due to case (in)sensitive filesystems mismatching between Darwin and linux 😞

Reply:
:/

Is there a tool to generate the hash from linux?

Reply:
Ofborg already did? :)

Reply:
Ah right, so now only adding that hash and a switch between those two possible hashes. Will add a commit tomorrow morning addressing this. 

Thanks for the help. :3

Reply:
Hash mismatch due to case insensitive + case preserving filesystems like darwin's and windows' would only happen if there were multiple files only differing in case, like a `Config` file and a `config` file. Might be worth double checking this isn't an issue. If you have access to a macOS machine you can do that by creating two volumes, a case insensitive one and a case sensitive one, and then diffing the extracted github tarball. On linux you can find ways of doing that on SO, like https://stackoverflow.com/questions/2109056/how-to-find-duplicate-files-with-same-name-but-in-different-case-that-exist-in-s for example.

Reply:
Looks like there are two identical files called `DrawTest.java` at nearly identical paths.

```
❯ curl -fsSL https://github.com/JetBrains/JetBrainsRuntime/archive/jb17.0.3-b469.32.tar.gz \
  | tar tz \
  | sort -f | uniq -D -i
JetBrainsRuntime-jb17.0.3-b469.32/test/jdk/jb/java/awt/font/
JetBrainsRuntime-jb17.0.3-b469.32/test/jdk/jb/java/awt/Font/
JetBrainsRuntime-jb17.0.3-b469.32/test/jdk/jb/java/awt/font/TextLayout/
JetBrainsRuntime-jb17.0.3-b469.32/test/jdk/jb/java/awt/Font/TextLayout/
JetBrainsRuntime-jb17.0.3-b469.32/test/jdk/jb/java/awt/font/TextLayout/DrawTest.java
JetBrainsRuntime-jb17.0.3-b469.32/test/jdk/jb/java/awt/Font/TextLayout/DrawTest.java
```

Reply:
`diff ./test/jdk/jb/java/awt/font/TextLayout/DrawTest.java test/jdk/jb/java/awt/Font/TextLayout/DrawTest.java` is empty, then an easier approach would be to delete one of the duplicates?

Reply:
Opened an issue upstream, going to go with the `hash = if macos then X else Y` approach.